BlockServerToKafka/kafka_producer.py

```diff
@@ -46,17 +46,19 @@ def _set_up_producer(self, server: str):
                 print_and_log(f"WARNING: topic {self.topic} does not exist. It will be created by default.")
         except errors.NoBrokersAvailable:
             print_and_log(f"No brokers found on server: {server[0]}")
-            quit()
         except errors.ConnectionError:
             print_and_log("No server found, connection error")
-            quit()
         except errors.InvalidConfigurationError:
             print_and_log("Invalid configuration")
             quit()
         except errors.InvalidTopicError:
             print_and_log("Invalid topic, to enable auto creation of topics set"
                           " auto.create.topics.enable to false in broker configuration")
-            quit()
+        finally:
+            print_and_log("Retrying in 10s")
+            sleep(10)
+            # Recursive call after waiting
+            self._set_up_producer(server)
 
```

Comment:
It should hit the limit of 1000 in about 7 days so should restart every 7 days, probably not the best fix in the world but hopefully will reduce symptoms until the cluster is back up. We could retry forever or give up after a few attempts, but leave the process running instead of using quit() so that it doesn't restart?